### PR TITLE
[DOCS] Additional explanation for ConfiguredAssetDataConnector and InferredAssetDataConnector docs

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
Changes proposed in this pull request:
-  how-to-docs make distinction between InferredAssetDataConnector and its subclasses : `InferredAssetFilesystemDataConnector` and `InferredAssetS3DataConnector`.
- how-to-docs make distinction between ConfiguredAssetDataConnector and its subclasses : `ConfiguredAssetFilesystemDataConnector` and `ConfiguredAssetS3DataConnector`.
- changed reference names for `How to configure an InferredAssetDataConnector` and `How to configure an ConfiguredAssetDataConnector`
- added section on `Choose a DataConnector` which (hopefully) adds useful explanation before more detailed examples.  
- added blurb mentioning how `group_names` can be used by `context.get_batch()`

### CODE REVIEW

File: docs/guides/how_to_guides/configuring_datasources/how_to_configure_a_configuredassetdataconnector.rst [hunk 1]
```diff
@@ -31,23 +30,95 @@ All of the examples below assume you’re testing configuration using something
...
+    my_batch = context.get_batch(
+        datasource_name="my_data_source",
+        data_connector_name="my_filesystem_data_connector",
+        data_asset_name="alpha",
+        partition_identifiers={"index": "2"}
+        )
+
+
+This ability to access specific partitions using ``partition_identifiers`` is very useful when validating DataAssets that span multiple files.
+For more information on ``batches`` and ``partition_identifiers``, please refer to the :ref:`Core Concepts document. <reference__core_concepts>`
```
Comment (Shinnnyshinshin) [inline docs/guides/how_to_guides/configuring_datasources/how_to_configure_a_configuredassetdataconnector.rst:98]: Not sure if this belongs here.. should partition_identifiers, splitting, sampling etc be in their own how-to?

File: docs/guides/how_to_guides/configuring_datasources/how_to_configure_a_configuredassetdataconnector.rst [hunk 2]
```diff
@@ -143,22 +216,21 @@ Then this configuration...
 
     class_name: ConfiguredAssetFilesystemDataConnector
     base_directory: test_data/
-    default_regex:
     assets:
         alpha:
             group_names:
-                - data_asset_name
+                - name
```
Comment (Shinnnyshinshin) [inline docs/guides/how_to_guides/configuring_datasources/how_to_configure_a_configuredassetdataconnector.rst:222]: to lessen confusion with the `InferredAssetFilesystemDataConnector`

**Review by eugmandel — APPROVED** [review]
Looks great!

**Comment by eugmandel** [comment]
This really helped me understand these connectors!

**Comment by Shinnnyshinshin** [comment]
Ok thank you @eugmandel will merge once pipelines pass

**Pull request merged**